### PR TITLE
Patch irrlicht for macOS Mojave and add bottle

### PR DESCRIPTION
**Pull request by kevinlul**
To allow building from source, replace the user-triggered error with a warning for now, until irrlicht updates. Hopefully, the edge case of broken PNGs isn't a big enough problem to warrant an entire broken build.

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/irrlicht.rb [hunk 1]
```diff
@@ -7,6 +7,7 @@ class Irrlicht < Formula
 
   bottle do
     cellar :any_skip_relocation
+    sha256 "88ef6a24a237c71b1342f4cd6a8dbf6d797d2460b1020cdf7f49b7ef0d2a24a2" => :mojave
```
Comment (mistydemeo) [inline Formula/irrlicht.rb]: Can you please remove this line? A bottle will be automatically built when we merge it. Thanks!

File: Formula/irrlicht.rb [hunk 2]
```diff
@@ -24,6 +25,12 @@ def install
       "[NSApp setDelegate:(id<NSFileManagerDelegate>)",
       "[NSApp setDelegate:(id<NSApplicationDelegate>)"
 
+    # Fix "error: ZLIB_VERNUM != PNG_ZLIB_VERNUM" on Mojave (picking up system zlib)
+    # Reported 21 Oct 2018 https://sourceforge.net/p/irrlicht/bugs/442/
+    inreplace "source/Irrlicht/libpng/pngpriv.h",
+      "#  error ZLIB_VERNUM != PNG_ZLIB_VERNUM \\",
+      "#  warning ZLIB_VERNUM != PNG_ZLIB_VERNUM \\"
+
```
Comment (mistydemeo) [inline Formula/irrlicht.rb]: Looks like there's some stray whitespace here; can you remove that?

**Review by mistydemeo — CHANGES_REQUESTED** [review]
Thanks for the submission! Left a couple of comments.

**Review by fxcoudert — APPROVED** [review]

**Comment by fxcoudert** [comment]
Thanks @kevinlul

**Pull request closed without merging**